driver-core/src/main/com/mongodb/internal/connection/ReadConcernHelper.java

```diff
@@ -35,12 +38,22 @@ public static BsonDocument getReadConcernDocument(final SessionContext sessionCo
             readConcernDocument.append("level", new BsonString(level.getValue()));
         }
 
+        assertFalse(sessionContext.isSnapshot() && sessionContext.isCausallyConsistent());
+        if (sessionContext.isSnapshot() && maxWireVersion < FIVE_DOT_ZERO_WIRE_VERSION) {
+            throw new MongoClientException("Snapshot reads require MongoDB 5.0 or later");
```

Comment:
Add test to ensure this is handled in reactive streams

Reply:
There is a spec test for this in "snapshot-sessions-not-supported-client-error", and a reactive streams runner for sessions tests.  So I think it's covered already?

Reply:
I just double checked and it was all good.